src/testsets.jl

```diff
@@ -28,56 +24,81 @@ function finish(ts::ConcurrentTestSet)
     if Test.get_testset_depth() > 0
         # Attach this test set to the parent test set
         parent_ts = Test.get_testset()
-        record(parent_ts, ts.dts)
+        record(parent_ts, ts)
```

Comment:
Does this work? I think ConcurrentTestSet doesn't implement all the methods needed for this. (Although maybe it should.)

Reply:
It does, as I added the methods